### ISSUE TITLE
kpr: stabilize Service handling for loopback connections

### ISSUE THREAD

**Issue by julianwiedmann** [cilium/cilium#33194]
The handling for IPv4 Service loopback connections in the per-packet LB (client connects to itself via a service) is a tricky beast. https://github.com/cilium/cilium/pull/32480 improved the situation quite a bit, but there's still some work to do:

* [x] add connectivity tests to catch regressions
* [x] add support for IPv6 loopback (https://github.com/cilium/cilium/issues/26733)
    - Ongoing PR at https://github.com/cilium/cilium/pull/39594.
* [ ] document the special handling (for instance: emphasize that no network policy is needed)

**Comment by aditighag** [comment]
>add connectivity tests to catch regressions

We have e2e ginkgo tests to validate this packet path in the presence of L3 and L4 policies (see this commit for more context) - https://github.com/cilium/cilium/commit/3f9d44ecdd. However, the test also needs to be run for the endpoint routes mode.

**Comment by saiaunghlyanhtet** [comment]
@julianwiedmann 
Raise PR (but in draft). Can you give a look that I am on the right track. https://github.com/cilium/cilium/pull/38105

**Comment by julianwiedmann** [comment]
For the connectivity tests, we can possibly re-use some of the recent work by @sayboras for loopback via L7 service: https://github.com/cilium/cilium/pull/39577

**Comment by julianwiedmann** [comment]
@saiaunghlyanhtet :wave: just checking, are you still looking into the needed connectivity tests? It's a bit discouraging that [trying](https://github.com/cilium/cilium/pull/42439) to enable the IPv6 version of the existing test fails ... it's probably just a problem in the legacy test framework, but I really don't want to chase that down 🙂.

**Comment by saiaunghlyanhtet** [comment]
> [@saiaunghlyanhtet](https://github.com/saiaunghlyanhtet) 👋 just checking, are you still looking into the needed connectivity tests? It's a bit discouraging that [trying](https://github.com/cilium/cilium/pull/42439) to enable the IPv6 version of the existing test fails ... it's probably just a problem in the legacy test framework, but I really don't want to chase that down 🙂.

Yes, I am still working on it but as I changed my development laptop, I forgot to push that into git. I am so re-working on it.

**Comment by saiaunghlyanhtet** [comment]
@julianwiedmann 
I found the problem. Please check this [issue](https://github.com/cilium/cilium/issues/42460)

@pchaigno 
I think you also need to check the above issue as it is related to LLA  (fe80::1) we used in my previous pr for service loopback ipv6.

**Comment by julianwiedmann** [comment]
> [@saiaunghlyanhtet](https://github.com/saiaunghlyanhtet) 👋 just checking, are you still looking into the needed connectivity tests? It's a bit discouraging that [trying](https://github.com/cilium/cilium/pull/42439) to enable the IPv6 version of the existing test fails ... it's probably just a problem in the legacy test framework, but I really don't want to chase that down 🙂.

Rebased https://github.com/cilium/cilium/pull/42439 on-top of https://github.com/cilium/cilium/pull/42497, but still no luck.

**Comment by saiaunghlyanhtet** [comment]
@julianwiedmann sorry for late reply. I don't find any problem in connectivity tests. But, maybe tests are not covering enough. I have created a new pr.

**Comment by saiaunghlyanhtet** [comment]
@julianwiedmann 
Conn test pr is merged. But, I think we need to add something for docs. Unfortunately, I am not that good with docs.
